### ISSUE TITLE
Complete names inside deconstruction pattern

### ISSUE THREAD

**Issue by jcouv** [dotnet/roslyn#26619]
There could also be a completion provider for name: inside a deconstruction pattern, similar to the one for property patterns. (https://github.com/dotnet/roslyn/pull/26553#issuecomment-386470879)

That said, I don't expect usage of names in deconstructions to be common. It may be good to wait for some real-world experience.

Example:
```c#
class C
{
    void Deconstruct(out int first, out int second) => throw null;
    void M()
    {
        _ = this is ($$ 1, 2); // should suggest "first" or maybe "first:"
   }
}
```

**Comment by Neme12** [comment]
Note: this should behave very much like the completion provider for named arguments (possibly even be part of it). That one shows `:` in the completion list but doesn't actually insert it on commit:
![image](https://user-images.githubusercontent.com/11444821/39608802-5efeb510-4f44-11e8-9dab-b3dcc5c17967.png)

**Comment by Neme12** [comment]
I agree that this might not be very common but I expect completion to work nonetheless.

**Comment by CyrusNajmabadi** [comment]
Closing out due to lack of feedback on this.
